tests/ui/src/utilities/cloudFront.ts

```diff
@@ -1,5 +1,32 @@
-import childProcess from "child_process";
+import { CloudFrontClient, ListDistributionsCommand } from '@aws-sdk/client-cloudfront';
 
-export function getCloudFrontUrl(region: string, env: string, workspace: string) {return childProcess.execSync(
-  `aws cloudfront list-distributions --query "DistributionList.Items[].{DomainName: DomainName, OriginDomainName: Origins.Items[0].DomainName}[?OriginDomainName=='nhse-uec-cm-`+env+`-front-end`+workspace+`.s3.`+region+`.amazonaws.com'] | [0]"`).toString();
+export async function getCloudFrontUrl(region: string, env: string, workspace: string) {
+  const client = new CloudFrontClient({ region: region });
+
+  try {
+    const command = new ListDistributionsCommand({});
+    const response = await client.send(command);
+
+    if (response.DistributionList && response.DistributionList.Items) {
+      const distributions = response.DistributionList.Items;
+      const filteredDistribution = distributions.find(dist =>
+        dist.Origins?.Items?.[0].DomainName === `nhse-uec-cm-${env}-front-end${workspace}.s3.${region}.amazonaws.com`
+      );
+
+      if (filteredDistribution) {
+        return {
+          DomainName: filteredDistribution.DomainName,
+        };
+      } else {
+        console.log('No matching distribution found');
+        return null;
+      }
+    } else {
+      console.log('No distributions found');
+      return null;
+    }
```

Comment:
Nitpick/Suggestion - using negations instead can help make your code clearer and less nested. Also - if we just need the domain name, I'd return that directly.

 For example:
```suggestion
    const distributions = response.DistributionList?.Items;
    
    if (!distributions) {
        console.log('No distributions found');
        return null;
    }
    
    const filteredDistribution = distributions.find(dist =>
        dist.Origins?.Items?.[0].DomainName === `nhse-uec-cm-${env}-front-end${workspace}.s3.${region}.amazonaws.com`
    );
    
    if (!filteredDistribution) {
        console.log('No matching distribution found');
        return null;
    }
    
    return filteredDistribution.DomainName;
```